superset/datasets/api.py

```diff
@@ -769,9 +775,15 @@ def import_(self) -> Response:
             else None
         )
         overwrite = request.form.get("overwrite") == "true"
+        sync_columns = request.form.get("sync_columns") == "true"
+        sync_metrics = request.form.get("sync_metrics") == "true"
 
         command = ImportDatasetsCommand(
-            contents, passwords=passwords, overwrite=overwrite
+            contents,
+            passwords=passwords,
+            overwrite=overwrite,
+            sync_columns=sync_columns,
+            sync_metrics=sync_metrics,
```

Comment:
Side note, I see that V0.ImportDatasetsCommand will use the sync kwargs but not V1.

Reply:
cc @betodealmeida is that intentional? ☝️

Reply:
It appears that V1 uses the overwrite flag to denote syncing columns and metrics.

Reply:
I'm a bit confused here... passing `sync_columns` and `sync_metrics` to `ImportDatasetsCommand` has no effect, as far as I can see. Can you explain more the use case for this, @reesercollins?

Reply:
`sync_columns` and `sync_metrics` means the import command will remove columns and metrics which don't appear in the dataset YAML. https://github.com/apache/superset/blob/76d6a9af91a71401d158bc150518b3c9366030ac/superset/datasets/commands/importers/v0.py#L249

Reply:
Ah, that makes sense, it's for the v0 API! Yeah, in this case it should be fine... the v1 API will ignore these extra arguments, and should be fine.

But then my next question is... any reason why you're still using the v0 API, instead of enabling the `VERSIONED_EXPORT` feature flag? The v0 import IMHO is very buggy, the code is incomplete and non-standard across asset types, and we should deprecated it as soon as possible.

Reply:
We have some automation we haven't gotten around to updating which generates the YAML files. We very rarely do any exporting.